### PR TITLE
@inline getindex and setindex! for Field

### PR DESCRIPTION
**Pull request by asapabedi**
Will see!

### CODE REVIEW

File: src/fields.jl [hunk 1]
```diff
@@ -116,8 +116,8 @@ location(::Field{Lx, Ly, Lz}) where {Lx, Ly, Lz} = (Lx, Ly, Lz)
 @inline size(f::AbstractField) = size(f.grid)
 @inline length(f::AbstractField) = length(f.data)
 
-@propagate_inbounds getindex(f::AbstractField, inds...) = getindex(f.data, inds...)
-@propagate_inbounds setindex!(f::AbstractField, v, inds...) = setindex!(f.data, v, inds...)
+@inline @propagate_inbounds getindex(f::AbstractField, inds...) = getindex(f.data, inds...)
```
Comment (glwagner) [inline src/fields.jl]: I think that `@propagate_inbounds` implies `@inline`, which means that the `@inline` annotation is redundant here. Is there a demonstrated performance benefit from this change? If so, why?

**Comment by ali-ramadhan** [comment]
Welcome @asapabedi thank you for opening this PR! I meant to merge it earlier.

@glwagner That's what I thought but it seems that with both macros there is a very slight performance improvement.

See benchmarks at: https://github.com/thabbott/JULES.jl/pull/22

The point of this was to finally close issue #13

**Comment by ali-ramadhan** [comment]
@asapabedi in this case opening this pull request was useful (thank you!), but in general it can be a bad idea to open pull requests using other people's branches and code.

They might be in the middle of working on an issue and the branch might not be ready to be merged. In general you shouldn't touch other people's branches before asking.

**Comment by asapabedi** [comment]
Clearly stated. Thanks

**Comment by glwagner** [comment]
> The point of this was to finally close issue #13

I believe issue #13 occurs when `@inbounds` is not used. I found this same issue (and its resolution) a while ago during the development of `OceanTurb.jl`.

**Comment by ali-ramadhan** [comment]
Yeah I thought it might correspond to a slight measurable improvement as `@benchmark` averaged over ~2700 calls, but maybe many more calls are needed before to determine whether there is any difference.

Either way, a difference that small probably won't matter.

**Comment by glwagner** [comment]
I think we should close this PR.

**Pull request closed without merging**